### PR TITLE
chore: Make CoreSocketFactory a real singleton.

### PR DESCRIPTION
**Pull request by hessjcg**
Create the singleton instance of CoreSocketFactory when the class is loaded instead of a lazy instantiation.

### CODE REVIEW

File: core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java [hunk 1]
```diff
@@ -71,14 +71,18 @@ public final class CoreSocketFactory {
...
+  private static final CoreSocketFactory coreSocketFactory;
   private final ListenableFuture<KeyPair> localKeyPair;
   private final ConcurrentHashMap<String, CloudSqlInstance> instances = new ConcurrentHashMap<>();
   private final ListeningScheduledExecutorService executor;
   private final CredentialFactory credentialFactory;
   private final int serverProxyPort;
   private final SqlAdminApiFetcher adminApiService;
 
+  static {
+    coreSocketFactory = newFactory();
```
Comment (enocom) [inline core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java:83]: Shall we use an enum instead?

https://stackoverflow.com/questions/70689/what-is-an-efficient-way-to-implement-a-singleton-pattern-in-java

Comment (enocom) [reply core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java:83]: In that case, what do you see as the benefit of this change?

Comment (hessjcg) [reply core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java:83]: It's a step towards getting rid of the KeyPair future, which is coming in the next few PRs.

**Comment by hessjcg** [comment]
Closing, will combine with a different PR.

**Pull request closed without merging**